src/organization/screens/repository-list.screen.js

```diff
@@ -0,0 +1,223 @@
+/* eslint-disable no-shadow */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { FlatList, View, Dimensions, StyleSheet } from 'react-native';
+
+import {
+  ViewContainer,
+  RepoListItem,
+  LoadingRepositoryListItem,
+  SearchBar,
+} from 'components';
+import { colors } from 'config';
+
+import { getRepos } from 'api/rest/providers/github/endpoints/orgs';
+import { searchRepos } from 'api/rest/providers/github/endpoints/search';
+
+const loadData = ({ orgId, getRepos }) => {
+  getRepos(orgId);
+};
+
+// TODO: clean up searchStart & query usage
+
+const styles = StyleSheet.create({
+  header: {
+    borderBottomColor: colors.greyLight,
+    borderBottomWidth: 1,
+  },
+  searchBarWrapper: {
+    flexDirection: 'row',
+  },
+  searchContainer: {
+    width: Dimensions.get('window').width,
+    backgroundColor: colors.white,
+    flex: 1,
+  },
+  searchCancelButton: {
+    color: colors.black,
+  },
+  listContainer: {
+    marginBottom: 90,
+  },
+});
+
+class OrgRepositoryList extends Component {
+  props: {
+    searchRepos: Function,
+    getRepos: Function,
+    orgId: String,
+
+    searchedResults: Array,
+    repositories: Array,
+
+    repositoriesPagination: Object,
+    searchedResultsPagination: Object,
+
+    authUser: Object,
+
+    navigation: Object,
+  };
+
+  state: {
+    query: string,
+    searchStart: boolean,
+    searchFocus: boolean,
+  };
+
+  constructor() {
+    super();
+
+    this.state = {
+      query: '',
+      searchStart: false,
+      searchFocus: false,
+    };
+
+    this.search = this.search.bind(this);
+    this.getList = this.getList.bind(this);
+  }
+
+  componentWillMount() {
+    loadData(this.props);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.orgId !== this.props.orgId) {
+      loadData(nextProps);
+    }
+  }
+
+  getList = () => {
+    const { searchedResults, repositories } = this.props;
+    const { searchStart } = this.state;
+
+    return searchStart ? searchedResults : repositories;
+  };
+
+  loadMore = () => {
+    const { orgId, getRepos, searchRepos, navigation } = this.props;
+
+    if (navigation.state.params.searchedKeyword) {
+      searchRepos(navigation.state.params.searchedKeyword, true);
+    } else {
+      getRepos(orgId, { loadMore: true });
+    }
+  };
+
+  search(query) {
+    if (query !== '') {
+      const searchedKeyword = `q=${query}+user:${this.props.navigation.state
+        .params.orgId}+fork:true`;
+
+      this.setState({
+        searchStart: true,
+        query,
+      });
+
+      const { searchRepos, navigation } = this.props;
+
+      navigation.setParams({ searchedKeyword });
+      searchRepos(searchedKeyword);
```

Comment:
I used `navigation.setParams` to store the searchedKeyword, later used in `mapStateToProps()`.
Is that okay, or considered a bad practice?